File: C++/ReversingQueue.cpp
#include <bits/stdc++.h>
using namespace std;
 
void Print(queue<int>& Queue)
{
    while (!Queue.empty()) {
        cout << Queue.front() << " ";
        Queue.pop();
    }
}
 
void reverseQueue(queue<int>& Queue)
{
    stack<int> Stack;
    while (!Queue.empty()) {
        Stack.push(Queue.front());
        Queue.pop();
    }
    while (!Stack.empty()) {
        Queue.push(Stack.top());
        Stack.pop();
    }
}
 
int main()
{
    queue<int> Queue;
    Queue.push(10);
    Queue.push(20);
    Queue.push(30);
    Queue.push(40);
    Queue.push(50);
    Queue.push(60);
    Queue.push(70);
    Queue.push(80);
    Queue.push(90);
    Queue.push(100);
 
    reverseQueue(Queue);
    Print(Queue);
}

File: Python/ReversingQueue.py
from queue import Queue

def Print(q):
    print(q.queue)

def reverse(queue):
    stack = []
    while not queue.empty():
        stack.append(queue.queue[0])
        queue.get()
    while len(stack) != 0:
        queue.put(stack[-1])
        stack.pop()
 
if __name__ == '__main__':
    queue = Queue()
    items = [1,3,5,7,9,11]
    for i in items:
        queue.put(i)
    reverse(queue)
    Print(queue)